R/examine-constraints.R

```diff
@@ -135,11 +135,20 @@ check_fk_constraints <- function(dm, progress = NA) {
 
   ticker <- new_ticker("checking pk constraints", nrow(fks_tibble), progress = progress)
 
+  problems <-
+    pmap(fks_tibble, ticker(check_fk), fk_repair = fk_repair) %>%
+    transpose()
```

Comment:
```suggestion
    pmap(fks_tibble, ticker(check_fk), fk_repair = fk_repair, sample = sample)
```

Can we derive the label from the repair plan here?

Reply:
At the moment we might, but our repair plan for pruning doesn't work, if for example we end up with a repair plan containing the subset of the fact table to be removed, it wouldn't be possible anymore. Unless we make the repair plan a richer object that would contain the ingredients to build the label, I don't know if it's more elegant than the current.

Reply:
I think `dm_rows_delete()` should use all columns if no key column is set. Also, I think we should change it to take the PK information from the "other" dm (that contains the rows to be deleted). We can handle that in a separate PR that we merge prior to this PR.